src/app.js

```diff
@@ -1,7 +1,20 @@
 /* @flow */
 'use strict'
 
+// SIGINT and uncaughtException do not run cleanup
+// so we have an exitHandler to clean up the database connection
+function exitHandler (error) {
+  if (error) {
+    console.error('Exiting: ', error)
+    process.exit(1)
+  }
+  process.exit(0)
+}
+
 module.exports = require('./lib/app')
 if (!module.parent) {
+  process.on('SIGINT', exitHandler)
+  process.on('uncaughtException', exitHandler)
```

Comment:
I've just tested again locally.

1) Calling knex.destroy() in process.on('exit') doesn't do anything. The documentation states, "Therefore you must only perform synchronous operations in this handler." knex.destroy is async

2) If the process handles SIGINT and doesn't exit, it must be sent a SIGKILL (`kill -9`)

This must be sent a SIGKILL for it to exit, ctrl+c won't work.
```
process.on('SIGINT', () => {})
setTimeout(() => {}, 1000000)
```